src/charm.py

```diff
@@ -117,19 +110,23 @@ def _ensure_kube_control_relation(self, event: EventBase) -> bool:
         )
         return True
 
-    def _ensure_certificates_relation(self, event: EventBase) -> bool:
-        self.unit.status = MaintenanceStatus("Evaluating certificates.")
+    def _ensure_certificates_relation(self, event: ops.EventBase) -> bool:
+        if self.kube_control.get_ca_certificate():
+            logger.info("CA Certificate is available from kube-control.")
+            return True
```

Comment:
If we get a CA certificate from the `kube-control` relation, we don’t need to write the CA cert from the TLS relation, right? However, I notice we are using `CA_CERT_PATH` to create the kubeconfig (L112,L115). Am I missing something here? Could you clarify if I’m misunderstanding this?

Reply:
details from the library:


```python
    def create_kubeconfig(
        self, ca: PathLike, kubeconfig: PathLike, user: str, k8s_user: str
    ):
        ... Skipped ...
        if ca_content := self.get_ca_certificate():
            ca_b64 = base64.b64encode(ca_content).decode("utf-8")
        elif Path(ca).exists():
            ca_b64 = base64.b64encode(Path(ca).read_bytes()).decode("utf-8")
        else:
            log.error("No CA certificate found")
            raise FileNotFoundError("No CA certificate found")
```